utils/makefile.python

```diff
@@ -119,13 +120,13 @@ quiet_cmd_pyenvuninstall = PYENV     uninstall   $2
 # $2 path to folder where virtualenv take place
 quiet_cmd_virtualenv  = PYENV     usage: $ source ./$@/bin/activate
       cmd_virtualenv  = \
-	if [ ! -d "./$(PY_ENV)" ];then                                  \
-		$(PYTHON) -m venv $(VTENV_OPTS) $2;                     \
+	if [ -d "./$(PY_ENV)" -a -x "./$(PY_ENV_BIN)/python" ]; then \
```

Comment:
Checking the existence of `/$(PY_ENV_BIN)/python"` makes also sense to me.  This might help in cases where a previous build of the virtualenv has been failed for any reason.  

I thought twice about a more robust criteria but I don't have anything better to suggest.

Reply:
Another thought:  now, with removing ` $(PY_ENV)/requirements.sha256` before running `pip install` and the `&& sha256sum requirements*.txt > $(PY_ENV)/requirements.sha256` from line 104-105, checking existence of  ` $(PY_ENV)/requirements.sha256`  could be a good criteria here?

Reply:
Is there a case where the `pyenv` target is invoked but the `pyenvinstall` target ?
* if no, why there are two targets ?
* if yes, then we _can't_ `$(PY_ENV)/requirements.sha256` : each invocation of `pyenv` will reinstall the virtualenv.

Reply:
These makefiles has been implemented for general use cases.  I use them in several projects as [boilerplate](https://return42.github.io/boilerplate/slides/index.html#/).  The motivation was to have  typical development workflows localy at hand, build on simple tools available nearly everywhere.

Back to your question _why there are two targets ?_ 

pyenv: builds the environment for development / typically `requirements.txt`
penvinstall: installs the python package in this environment (`pip install -e`) / typically `setup.py` 

In the searx project `requirements.txt` and `setup.py` include identical _requirements_ but this is not typical for python projects.  Beside this you can also think about projects containing more than one package.

> Is there a case where the pyenv target is invoked but not the pyenvinstall target ?

With the words above, my answer is: _it depends .._ But from `makefile.python`'s POV the answer is : yes

Your questions helped me to sort my thoughts, but for today I am to tired .. tomorrow I have more time, then I'll think about it again ..

- pyyaml in setup.py
- _each invocation of pyenv will reinstall the virtualenv_